### PR TITLE
Deep merge SageMaker `additional_args`, allowing more flexible configuration and `env` variable support

### PR DESCRIPTION
**Pull request by dbpprt**
The current implementation allows to naively override parameters set by accelerate, which breaks the code in the following case:

`user wants to set custom environment variables`
```python
args = {
    "entrypoint": "train.py",
    "environment": {
        # just an example
        "PYTHONPATH": "/opt/ml/code",
        "SAGEMAKER_PROGRAM": "train.py",
        "SAGEMAKER_SUBMIT_DIRECTORY": "s3://sagemaker-us-east-1-123456789012/sagemaker-pytorch-2020-12-01-20-00-00-123/source/sourcedir.tar.gz",
    }
}

additional_args = {
    "use_spot": True,
    "environment": {
        "ADDITIONAL_ENV": "1234567890",
    },
}

{**args, **additional_args}
```

`environment` is replaced completely, which breaks `accelerate`.
```
{'entrypoint': 'train.py',
 'environment': {'ADDITIONAL_ENV': '1234567890'},
 'use_spot': True}
```

This PR allows both `dicts` to be deep-merged:
```python
merge_dicts(additional_args, args)
```

```
{'entrypoint': 'train.py',
 'environment': {'PYTHONPATH': '[/opt/ml/code](https://file+.vscode-resource.vscode-cdn.net/opt/ml/code)',
  'SAGEMAKER_PROGRAM': 'train.py',
  'SAGEMAKER_SUBMIT_DIRECTORY': 's3:[//sagemaker-us-east-1-123456789012/sagemaker-pytorch-2020-12-01-20-00-00-123/source/sourcedir.tar.gz](https://sagemaker-us-east-1-123456789012/sagemaker-pytorch-2020-12-01-20-00-00-123/source/sourcedir.tar.gz)',
  'ADDITIONAL_ENV': '1234567890'},
 'use_spot': True}
```

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks for your PR, LGTM! @pacman100 could you have a second look?

File: src/accelerate/commands/launch.py [hunk 1]
```diff
@@ -1062,7 +1063,7 @@ def sagemaker_launcher(sagemaker_config: SageMakerConfig, args):
     }
 
     if sagemaker_config.additional_args is not None:
-        args = {**args, **sagemaker_config.additional_args}
+        args = merge_dicts(sagemaker_config.additional_args, args)
```
Comment (pacman100) [inline src/accelerate/commands/launch.py:1066]: won't `args.update(sagemaker_config.additional_args)` do the trick?

**Review by pacman100 — COMMENTED** [review]
Hello, Thank you for the PR! Left a comment

**Comment by dennisbappert** [comment]
Thanks for the feedback, anything else required for this PR to get merged?

**Pull request merged**